index.js

```diff
@@ -128,60 +153,97 @@ async function assumeRole(params) {
-  const nameWithoutSpecialCharacters = name.replace(/[^\p{L}\p{Z}\p{N}_:/=+.-@-]/gu, SANITIZATION_CHARACTER);
-  const nameTruncated = nameWithoutSpecialCharacters.slice(0, MAX_TAG_VALUE_LENGTH)
-  return nameTruncated
+  const nameWithoutSpecialCharacters = name.replace(
+    /[^\p{L}\p{Z}\p{N}_:/=+.-@-]/gu,
+    SANITIZATION_CHARACTER
+  );
+  const nameTruncated = nameWithoutSpecialCharacters.slice(
+    0,
+    MAX_TAG_VALUE_LENGTH
+  );
+  return nameTruncated;
 }
 
-function exportCredentials(params){
+function exportCredentials(params) {
   // Configure the AWS CLI and AWS SDKs using environment variables and set them as secrets.
   // Setting the credentials as secrets masks them in Github Actions logs
-  const {accessKeyId, secretAccessKey, sessionToken} = params;
+  const { accessKeyId, secretAccessKey, sessionToken } = params;
 
   // AWS_ACCESS_KEY_ID:
   // Specifies an AWS access key associated with an IAM user or role
   core.setSecret(accessKeyId);
-  core.exportVariable('AWS_ACCESS_KEY_ID', accessKeyId);
+  core.exportVariable("AWS_ACCESS_KEY_ID", accessKeyId);
 
   // AWS_SECRET_ACCESS_KEY:
   // Specifies the secret key associated with the access key. This is essentially the "password" for the access key.
   core.setSecret(secretAccessKey);
-  core.exportVariable('AWS_SECRET_ACCESS_KEY', secretAccessKey);
+  core.exportVariable("AWS_SECRET_ACCESS_KEY", secretAccessKey);
 
   // AWS_SESSION_TOKEN:
   // Specifies the session token value that is required if you are using temporary security credentials.
   if (sessionToken) {
     core.setSecret(sessionToken);
-    core.exportVariable('AWS_SESSION_TOKEN', sessionToken);
+    core.exportVariable("AWS_SESSION_TOKEN", sessionToken);
   } else if (process.env.AWS_SESSION_TOKEN) {
     // clear session token from previous credentials action
-    core.exportVariable('AWS_SESSION_TOKEN', '');
+    core.exportVariable("AWS_SESSION_TOKEN", "");
   }
 }
 
+function exportCredentialsToConfig(profile, params) {
+  const awsConfigFile = "~/.aws/config";
+  if (fs.existsSync(awsConfigFile)) {
+    var config = ini.parse(fs.readFileSync(awsConfigFile, "utf-8"));
+  } else {
+    var config = {};
+  }
+
+  // Initialize the configuration object
+  const profileName = `profile ${profile}`;
+  config[profileName] = {};
+
+  // Configure the AWS CLI and AWS SDKs using environment variables and set them as secrets.
+  // Setting the credentials as secrets masks them in Github Actions logs
+  const { webIdentityTokenFile, roleArn, regionName } = params;
+
+  // web_identity_token_file:
+  // Specifies a web identity token file location
+  config[profileName].web_identity_token_file = webIdentityTokenFile;
+
+  // role_arn:
+  // Specifies the role to assume by this profile
+  config[profileName].role_arn = roleArn;
+
+  // region:
+  // Specifies the region for this profile
+  config[profileName].region = regionName;
+
+  fs.writeFileSync(awsConfigFile, ini.stringify(config));
+}
+
```

Comment:
This is the main change

Reply:
The config needs to point to a webIdentityToken file, which is already supported. We just need to point to it from multiple profiles.